test/jdk/java/awt/Mixing/AWT_Mixing/ViewportOverlapping.java

```diff
@@ -99,60 +103,75 @@ public void mouseClicked(MouseEvent e) {
+                hLoc = scrollPane.getHorizontalScrollBar().getLocationOnScreen();
+                hLoc.translate(scrollPane.getHorizontalScrollBar().getWidth() - 3, 3);
+                vLoc = scrollPane.getVerticalScrollBar().getLocationOnScreen();
+                vLoc.translate(3, scrollPane.getVerticalScrollBar().getHeight() - 3);
 
-                    testLoc = p.getLocationOnScreen();
-                    testLoc.translate(-3, -3);
+                testLoc = p.getLocationOnScreen();
+                testLoc.translate(-3, -3);
 
-                    resizeLoc = f.getLocationOnScreen();
-                    resizeLoc.translate(f.getWidth() - 1, f.getHeight() - 1);
-                }
+                resizeLoc = f.getLocationOnScreen();
+                resizeLoc.translate(f.getWidth() - 1, f.getHeight() - 1);
             });
         } catch (Exception e) {
             e.printStackTrace();
             throw new RuntimeException("Problem preparing test GUI.");
         }
-        // run robot
-        Robot robot = Util.createRobot();
-        robot.setAutoDelay(ROBOT_DELAY);
 
         robot.mouseMove(hLoc.x, hLoc.y);
-        robot.mousePress(InputEvent.BUTTON1_MASK);
-        robot.mouseRelease(InputEvent.BUTTON1_MASK);
-        Util.waitForIdle(robot);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        robot.waitForIdle();
+        captureScreen("Img_1.png");
 
         robot.mouseMove(vLoc.x, vLoc.y);
-        robot.mousePress(InputEvent.BUTTON1_MASK);
-        robot.mouseRelease(InputEvent.BUTTON1_MASK);
-        Util.waitForIdle(robot);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        robot.waitForIdle();
+        captureScreen("Img_2.png");
 
         clickAndBlink(robot, testLoc, false);
         robot.mouseMove(resizeLoc.x, resizeLoc.y);
-        robot.mousePress(InputEvent.BUTTON1_MASK);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
         robot.mouseMove(resizeLoc.x + 5, resizeLoc.y + 5);
-        robot.mouseRelease(InputEvent.BUTTON1_MASK);
-        Util.waitForIdle(robot);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        robot.waitForIdle();
+        captureScreen("Img_3.png");
 
         clickAndBlink(robot, testLoc, false);
-        return frameClicked == 2;
+        captureScreen("Img_4.png");
+        return (frameClicked == 2);
     }
 
     // this strange plumbing stuff is required due to "Standard Test Machinery" in base class
-    public static void main(String args[]) throws InterruptedException {
+    public static void main(String args[]) throws Exception {
+        robot = new Robot();
+        robot.setAutoDelay(ROBOT_DELAY);
+
         instance = new ViewportOverlapping();
         OverlappingTestBase.doMain(args);
+        captureScreen("Img_5.png");
+    }
```

Comment:
It is just a test, so no need to over-engineer it but  it 'feels wrong' to specify the file name
as foo.png here and separately inside captureScreen choose 'png' as the file type to use.
Perhaps  captureScreen("Img_5.png", "png")  ?

Reply:
Updated

Reply:
It doesn't feel right this way either. A cleaner approach would be to pass the undecorated file name and add the extension and specify the format inside `captureScreen`. (Passing the same value as a parameter usually produces an IDE warning, which means the parameter is not necessary.)

```java
captureScreen("Img_5");

ImageIO.write(
        robot.createScreenCapture(new Rectangle(0, 0, screenSize.width, screenSize.height)),
        "png",
        new File(filename + ".png")
);
```

Reply:
Updated.